webknossos/webknossos/dataset/dataset.py

```diff
@@ -1263,44 +1269,28 @@ def add_layer_from_images(
                 dtype=current_dtype,
             )
 
-            args = []
-            bbox = layer.bounding_box
-            additional_axes = [
-                axis_name for axis_name in bbox.axes if axis_name not in ("x", "y", "z")
-            ]
-            additional_axes_shapes = tuple(
-                product(
-                    *[range(bbox.get_shape(axis_name)) for axis_name in additional_axes]
+            if (
+                additional_axes := set(layer.bounding_box.axes).difference(
+                    "x", "y", "z"
                 )
-            )
-            if additional_axes and layer.data_format != DataFormat.Zarr3:
-                assert (
-                    len(additional_axes_shapes) == 1
+            ) and layer.data_format != DataFormat.Zarr3:
+                assert all(
+                    layer.bounding_box.get_shape(axis) == 1 for axis in additional_axes
                 ), "The data stores additional axes with shape bigger than 1. These are only supported by data format Zarr3."
 
                 # Convert NDBoundingBox to 3D BoundingBox
-                bbox = BoundingBox(
-                    bbox.topleft_xyz,
-                    bbox.size_xyz,
+                layer.bounding_box = BoundingBox(
```

Comment:
In the old version, I used the local `bbox` variable, because I converted it to a relative bounding box for generating the args.

The conversion to a 3d bbox is not related to the fix and not needed for chunking. It is used only if there are more than 3 dimensions in the pims image stack and the user selected WKW as output format. In this case, I check if the other dimensions have only a size of 1. If that is the case, I convert the ndbbox to a3d bbox and continue with the creation of the WKW dataset. Oterwise an error is thrown, because WKW does not support nd data.

Reply:
> Oterwise an error is thrown, because WKW does not support nd data.

just to clarify: do you mean the assertion error in line 1277 which would be thrown if additional axes exist? or do you mean that the wkw code that is called later would throw because of the nd bbox?
I'm just asking because I want to understand why the assignment to `layer.bounding_box` is new in this PR. did you notice that it didn't work before otherwise?

Reply:
I understand your confusion about this change. Currently, this code path is not used by any of our test data. It was meant as a workaround if the new expected_bbox has more than three axes, but it is flat in these dimensions, so it can still be converted to wkw. I think it had no impact at all, because it just changed the local variable and not `layer.bounding_box`, 
I think the conversion from nd to 3d bbox can be removed, and instead an Error should be raised if there are more than 3 axes. What is your opinion?

Reply:
Yes, your newest commit makes sense and simplifies things :+1: Just a suggestion about the wording:
I'd change "The data stores additional axes with shape bigger than 1" to "The data stores additional axes other than x, y and z".

Ah, and one related thought:
What happens if the image data contains x, y and t for example? Will the conversion to WKW still work?

Reply:
Currently, it depends. If it is actually recognized as x, y and t it will fail, as the `copy_to_view` method iterates over the `z`-axis. In most cases if there are 3 axes and their names are not delivered by pims it will succeed, and it is assumed that the names of the axes are x, y and z.

Reply:
Ok, this sounds like this was already like that prior to this PR. Therefore, it's fine for now :)